src/lib/distributions.ml

```diff
@@ -21,3 +21,31 @@ let standard_normal_pdf =
 
 let poisson_cdf ~mean k =
     Functions.gammaq (floor (k +. 1.0)) mean
+
+let ln_beta_pdf ~alpha ~beta =
+  if alpha <= 0.0 then raise (Invalid_argument "alpha") else
+    if beta <= 0.0 then raise (Invalid_argument "beta") else
+      let alpha_minus_1 = alpha -. 1.0 in
+      let beta_minus_1 = beta -. 1.0 in
+      let z = Functions.ln_beta_func alpha beta in fun x ->
+        if x < 0.0 || x > 1.0 then neg_infinity else
+          if x = 0.0 then
+            (if (alpha < 1.0) then raise (Invalid_argument "alpha") else neg_infinity)
+            else if x = 1.0 then (if (beta < 1.0) then raise (Invalid_argument "beta")
+          else neg_infinity) else
+            let logX = log x in
+            let log1_minus_x = log1p (-.x) in
+            alpha_minus_1 *. logX +. beta_minus_1 *. log1_minus_x -. z
+
+let beta_pdf ~alpha ~beta =
+  let ln_pdf = ln_beta_pdf ~alpha ~beta in
+  fun x -> let value = ln_pdf x in
+  if value = neg_infinity then 0.0 else exp value
+
+let beta_cdf ~alpha ~beta =
```

Comment:
Ok, fixed (exponent function was scope around too much of the expression) 
Diff here (w/some formatting changes too): 6abe05e12b481496412849213ba9e167a8bfde52